pennylane/compiler.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This subpackage provides the JIT compilation support for PennyLane.
+
+
+This subpackage exclusively functions as a wrapper for PennyLane's JIT compiler packages, without independently
+implementing any compiler itself. Currently, it supports the ``pennylane-catalyst`` package, with plans to
+incorporate additional packages in the near future.
+
+For any compiler packages seeking to be registered, it is imperative that they expose the 'entry_points' metadata
+under the designated group name: ``pennylane.compilers``.
+"""
+
+from typing import List
+from importlib import reload
+import dataclasses
+import pkg_resources
+
+
+@dataclasses.dataclass
+class AvailableCompilers:
+    """This contains data of installed PennyLane compiler packages"""
+
+    names_entrypoints = {}
+
+
+def _refresh_compilers():
+    """Scan installed PennyLane compiler pacakges to refresh the compilers
+    names and entry points
+    """
+
+    reload(pkg_resources)
+
+    # Refresh the list of compilers
+    AvailableCompilers.names_entrypoints = {}
+
+    # Iterator packages entry-points with the 'pennylane.compilers' group name
+    for entry in pkg_resources.iter_entry_points("pennylane.compilers"):
+        module_name = entry.module_name
+        # Only need name of the parent module
+        module_name = module_name.split(".")[0]
+
+        if module_name not in AvailableCompilers.names_entrypoints:
+            AvailableCompilers.names_entrypoints[module_name] = {}
+        AvailableCompilers.names_entrypoints[module_name][entry.name] = entry
```

Comment:
Yes, this will occur with supporting more than one compiler package. We're planning to add more compiler package support to PL in the future. You can check PR https://github.com/PennyLaneAI/catalyst/pull/321 for an example.